### DISCUSSION TITLE
V2 trips search is timing out

### DISCUSSION THREAD

**Discussion by boredland** [transitland/transitland#344] [Q&A] [answered]
wanted to try out the trips endpoint using `https://transit.land/api/v2/rest/routes/r-u0yjq-m43/trips?apikey={key}`, but the call is pending forever and eventually (minutes later) comes back with:
```json
{
  "message":"The upstream server is timing out"
}
```

**Comment by markuskurbel** [comment]
Hi,
I am hang the same issue with queries like: `https://transit.land/api/v2/rest/routes?lat={lat}&lon={lon}&radius={r}&api_key={key}` for all locations I have tested. Setting a limit also does not help. Any ideas?

**Comment by irees** [comment, marked as answer]
@boredland I've deployed a change that should help improve the performance of the trips query. It should no longer time out, and I have some additional optimizations that will come later. Note that you can also request trips for a specific day with the `service_date` parameter.

**Reply by boredland** [reply]
tyvm! great product you built there!
